### PR TITLE
handle single argument in ExecutionOutputLink(issue #1793)

### PR DESCRIPTION
**Pull request by noskill**
Fix for issue #1793

### CODE REVIEW

**Comment by ngeiswei** [comment]
@noskill cool, I'll try to test it today. 2 remarks
1. You should mention the issue it fixes in the comment otherwise github does register it (if you use the key word Fix it will even automatically close the related issue once I merge).
2. Make sure there is a unit test for it (in that instance you just need to re-enable the test referenced in issue #1793).

**Comment by noskill** [comment]
@ngeiswei Ok, got it.

**Comment by linas** [comment]
This can be merged, it looks OK to me.  ... err mostly OK. The unit test should be changed to verify that it works both with, and without the ListLink.

**Comment by ngeiswei** [comment]
Yes, thanks @linas that's exactly what I wanted to say. So @noskill could you add the following unit test
```c++
/**
 * Execute clear box with single argument (a link different than list)
 */
void ExecutionOutputUTest::test_clear_box_single_argument_link(void)
{
	logger().debug("BEGIN TEST: %s", __FUNCTION__);

	Handle X = an(VARIABLE_NODE, "$X"),
		A = an(CONCEPT_NODE, "A"),
		B = an(CONCEPT_NODE, "B"),
		id = al(LAMBDA_LINK, X, X),
		AB = al(INHERITANCE_LINK, A, B),
		exec = al(EXECUTION_OUTPUT_LINK, id, AB);

	Instantiator inst(&as);
	Handle expected = AB,
		result = HandleCast(inst.execute(exec));

	TS_ASSERT_EQUALS(expected, result);

	logger().debug("END TEST: %s", __FUNCTION__);
}
```
and fix your PR accordingly? Thanks!

**Comment by ngeiswei** [comment]
Oh and please try to follow the opencog dev standard (80 col max), see http://wiki.opencog.org/w/Development_standards. Thanks.

**Comment by noskill** [comment]
@ngeiswei I fixed the style, but new test_clear_box_single_argument_link test fails: I got result = (ConceptNode "A") 
I'll try to fix it.

**Comment by noskill** [comment]
New test passes now

File: opencog/atoms/execution/Instantiator.cc [hunk 1]
```diff
@@ -358,8 +358,10 @@ Handle Instantiator::walk_tree(const Handle& expr, bool silent)
 			{
 				args = beta_reduce(args, *_vmap);
 			}
-
-			const HandleSeq& oset(args->getOutgoingSet());
+			const Type& arg_type = args->get_type();
+			// unpack list link
+			const HandleSeq& oset(arg_type == LIST_LINK ? args->getOutgoingSet():
+															HandleSeq{args});
```
Comment (ngeiswei) [inline opencog/atoms/execution/Instantiator.cc]: That line is strangely aligned, that is why smart tabs (also called mixed tabs) exists! 

See https://www.emacswiki.org/emacs/SmartTabs or https://github.com/opencog/atomspace/pull/1643#discussion_r184526590 for more info.

Smart/Mixed tabs is not mandatory but it is recommended and is likely supported by your editor.

Comment (noskill) [reply opencog/atoms/execution/Instantiator.cc]: Thanks!

**Comment by noskill** [comment]
@ngeiswei Anything else to do here?

**Comment by ngeiswei** [comment]
That's perfect, thanks.

**Comment by ngeiswei** [comment]
BTW, @noskill, could you remove the branches from your account
https://github.com/noskill/atomspace/branches
that have been merged to the master (or are generally no longer useful)?
It makes it easier for others to go through your work. Thanks.

**Comment by noskill** [comment]
@ngeiswei Ok, i'll do

**Pull request merged**